constellations/gemini-admin/src/emails/accountInfo.js

```diff
@@ -0,0 +1,134 @@
+import { getFullName } from "../utils/userUtils.js";
+
+export class AccountInfo extends React.Component {
+    state = {
+        id: 0,
+        email: "",
+        users: [],
+        transactions: [],
+        selectedUserEmails: [],
+        active: false,
+    };
+
+    formatCurrency = (amount) => {
+        return new Intl.NumberFormat("en-US", {
+            style: "currency",
+            currency: "USD",
+        }).format(amount);
+    };
+
+    onCheckUser = (e, userId) => {
+        const users = this.props.users;
+        if (e.target.checked) {
+            users.map((user, index) => {
+                if (user.id == userId) {
+                    var email = this.state.selectedUserEmails;
+                    email.push(user.email);
+                    this.setState({
+                        selectedUserEmails: email,
+                    });
+                }
+            });
+        } else {
+            users.map((user, index) => {
+                if (user.id == userId) {
+                    var email = this.state.selectedUserEmails;
+                    email.splice(email.indexOf(user.email), 1);
+                    this.setState({
+                        selectedUserEmails: email,
+                    });
+                }
+            });
+        }
+    };
+
+    render = () => {
+        const id = this.props.id;
+        const email = this.props.email;
+        const users = this.props.users;
+        const transactions = this.props.transactions;
+        const name = this.props.name;
+
+        var emails = [];
+        users.map((user, index) => {
+            emails.push(user.email);
+        });
+
+        const userRows = users.map((user, index) => {
+            var status = user.isGuardian ? "(guardian" : "(student";
+            status += user.email == email ? ", primary contact)" : ")";
+
+            if (this.state.selectedUserEmails.length > 0) {
+                this.state.selectedUserEmails.map((existingEmail, index) => {
+                    if (emails.indexOf(existingEmail) == -1) {
+                        this.setState({
+                            selectedUserEmails: [],
+                        });
+                    }
+                });
+            }
+            return (
+                <div className="row" key={index}>
+                    <input
+                        type="checkbox"
+                        onChange={(e) => this.onCheckUser(e, user.id)}
+                    />
+                    <span className="column">{getFullName(user)}</span>
+                    <span className="column">{user.email}</span>
+                    <span className="column status">{status}</span>
+                </div>
+            );
+        });
+
+        var balance = 0;
+        transactions.map((transaction, index) => {
+            const amount = transaction.amount;
+            balance += parseInt(amount);
+        });
+
+        return (
+            <section id="account-info">
```

Comment:
This section isn't too large, so I think it would usually be okay to keep it as a single component. It will probably make your life a bit easier. But the other page will probably need to divide into multiple components, so if it's not too hard, I actually recommend splitting this component into two components as well to match code organization.

It would be nice when viewing this code and other EmailToProgram code, they have similar code organizations, so it should be easy to follow. For example, in this page, One component should be the AccountInformation, the other component is the GeneratedEmail section. In the other page, you'll probably have like three sections (ProgramInfo, SelectUserList, GeneratedEmail). Since there are so many components, you'll end up wanting to have them in different files.

So would probably even name your current folder to `emailPayment`. And when creating the other page, place all the stuff into the folder `emailProgram`.